### PR TITLE
ADBDEV-4661 ORCA does not fully support using CTE with replicated tables 

### PR DESCRIPTION
**Pull request by dkovalev1**
Add an extra test case for planning queries with ORCA with replicated tables
using CTE.

There was a well-known problem in ORCA with planning distributed queries with
replicated tables using CTE. It was addressed several times in both adb and
upstream (PRs 13833 - https://github.com/arenadata/gpdb/commit/cdd532fe03768cdba22b45852f7c33008ef431b6,
PR 14896, PR 13728, ADBDEV-2411), but only the upstream commit https://github.com/arenadata/gpdb/commit/24a54a79fec5f1a9471eae6e8f982c6e96d62503
which included in 6.27 provided a more complete solution.

This patch adds one more test case for the aforementioned problem, which was not
appeared in the tests but was solved. In the particular query in the test in 
this patch ORCA could not build a plan due to too heavy restrictions applied to 
CPhysicalSequence by using CDistributionSpecNonSingleton.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/69779

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1327345

**Comment by BenderArenadata** [comment]
Failed job Resource group isolation tests on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1327346

**Comment by BenderArenadata** [comment]
Failed job ORCA code linter on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1327347

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1327339

**Comment by BenderArenadata** [comment]
Failed job Regression tests with ORCA on ppc64le: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1327340

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/72880

File: src/test/regress/sql/rpt.sql [hunk 1]
```diff
@@ -601,6 +601,23 @@ drop table if exists t1;
...
+t1 as (select * from tbl1),
+t2 as (select a, b from tbl2)
+ select * from t1 p
+  join t2 r on p.b = r.b
+  join t2 r1 on p.c = r1.b;
+
 -- start_ignore
 drop schema rpt cascade;
 -- end_ignore
+
```
Comment (red1452) [inline src/test/regress/sql/rpt.sql]: extra empty line.

**Review by red1452 — COMMENTED** [review]
Make description, as it requires at https://wiki.adsw.io/books/adb/page/adb-trebovaniia-k-oformleniiu-patca

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/73063

**Review by red1452 — DISMISSED** [review]

**Comment by BenderArenadata** [comment]
Failed job Regression tests with Postgres on x86_64: https://gitlab.adsw.io/arenadata/github_mirroring/gpdb/-/jobs/1516930

File: src/test/regress/sql/rpt.sql [hunk 2]
```diff
@@ -601,6 +601,22 @@ drop table if exists t1;
...
+-- start_ignore
+drop table if exists tbl2;
+drop table if exists tbl1;
+-- end_ignore
+
+create table tbl2 (a numeric, b varchar(255), c numeric) distributed replicated;
+create table tbl1 (a bigserial, b varchar(15), c varchar(15)) distributed replicated;
+
+explain (analyze off, costs off, verbose off)
+with
```
Comment (bimboterminator1) [inline src/test/regress/sql/rpt.sql:616]: What about checking execution as well?

File: src/test/regress/sql/rpt.sql [hunk 3]
```diff
@@ -601,6 +601,22 @@ drop table if exists t1;
...
 
+-- start_ignore
+drop table if exists tbl2;
+drop table if exists tbl1;
+-- end_ignore
+
+create table tbl2 (a numeric, b varchar(255), c numeric) distributed replicated;
+create table tbl1 (a bigserial, b varchar(15), c varchar(15)) distributed replicated;
+
+explain (analyze off, costs off, verbose off)
```
Comment (bimboterminator1) [inline src/test/regress/sql/rpt.sql:615]: what does this test check? **please, write the comments in the tests at the line before -- start_ignore** . And in patch description your are talking about some problem in a really abstract way, add at least 1 sentence specifying the problem.

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/73556

File: src/test/regress/sql/rpt.sql (hunk 3 above)
Comment (dkovalev1) [reply src/test/regress/sql/rpt.sql:615]: done

File: src/test/regress/sql/rpt.sql (hunk 2 above)
Comment (dkovalev1) [reply src/test/regress/sql/rpt.sql:616]: Good point. Added.

**Review by bimboterminator1 — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/73564

**Review by red1452 — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/73672

**Pull request merged**